secret_handlers.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/RedHatInsights/sources-api-go/dao"
+	m "github.com/RedHatInsights/sources-api-go/model"
+	"github.com/RedHatInsights/sources-api-go/service"
+	"github.com/RedHatInsights/sources-api-go/util"
+	"github.com/labstack/echo/v4"
+	"gorm.io/datatypes"
+)
+
+var getSecretDao func(c echo.Context) (dao.SecretDao, error)
+
+func getSecretDaoWithTenant(c echo.Context) (dao.SecretDao, error) {
+	requestParams, err := dao.NewRequestParamsFromContext(c)
+	if err != nil {
+		return nil, err
+	}
+
+	return dao.GetSecretDao(requestParams), nil
+}
+
+func SecretCreate(c echo.Context) error {
+	secretDao, err := getSecretDao(c)
+	if err != nil {
+		return err
+	}
+
+	createInputRequest := struct {
+		m.AuthenticationCreateRequest
+		UserOwnership bool `json:"user_ownership"`
+	}{}
```

Comment:
Can we move this to `model/` and give it a name like `SecretCreateRequest`? 

Edit: we also don't want to allow all the same fields like a regular authentication (no resource_type/resource_id) so maybe a new struct would be better for the create struct at least.

Reply:
Also maybe `user_scoped` for the JSON name?

Reply:
It is much better to have SecretCreateRequest with only required field - that's also solve other things like validation as mentioned.

Thanks for suggestions!

> Also maybe user_scoped for the JSON name?

I used this name also for field in struct, I guess this field name and json name should same.